tests/lib/external/snapd-testing-tools/tools/os.query

```diff
@@ -73,19 +73,59 @@ is_core24() {
 }
 
 is_core_gt() {
-    compare_ubuntu "${1:-}" "-gt"
+    VERSION=$1
+    if [ -z "$VERSION" ]; then
+        echo "os.query: version id is expected"
+        exit 1
+    fi
+
+    # While ubuntu classic image is used to build a core image
```

Comment:
Do I understand correctly that the logic assumes that we build ubuntu-core-XX images on the corresponding ubuntu XX.04 system?

Is this something that we can guarantee? Should we fail if this is not the case?

Reply:
you understand correctly.

I'll update this function and remove that assumption because it is something that we should handle in snapd code.  I think we should use the corresponding ubuntu function before we have booted in the new core system. 

does it make sense?

Reply:
I think this change has to be added in a different pr because we need to change many places in snapd code which is not related to just uc24

Reply:
Yes, as long as you are aware of this and have a plan to follow up I'm happy.